scripts/gatkcondaenv.yml

```diff
@@ -17,21 +22,28 @@ dependencies:
 - xz=5.2.3=0
 - zlib=1.2.11=0
 - pip:
-  - argparse
   - bleach==1.5.0
+  - cycler==0.10.0
   - enum34==1.1.6
   - h5py==2.7.1
   - html5lib==0.9999999
+  - joblib==0.11
   - keras==2.1.1
   - markdown==2.6.9
-  - numpy==1.13.3
-  - pymc3==3.1
+  - matplotlib==2.1.0
+  - pandas==0.21.0
+  - patsy==0.4.1
   - protobuf==3.5.0.post1
+  - pymc3==3.1
+  - pyparsing==2.2.0
+  - python-dateutil==2.6.1
+  - pytz==2017.3
   - pyyaml==3.12
   - scipy==1.0.0
   - six==1.11.0
   - tensorflow==1.4.0
   - tensorflow-tensorboard==0.4.0rc3
   - theano==0.9.0
+  - tqdm==4.19.4
   - werkzeug==0.12.2
-
+  - "--editable=/gatk/src/main/python/org/broadinstitute/hellbender/tools/gcnv/"
```

Comment:
@samuelklee The env changes mostly seem fine (@lucidtronix - are you using agparse ?), except for the last line. Rather than the conda .yml be dependent on the repo structure, we should remove that last line, and instead rely on the changes implemented in https://github.com/broadinstitute/gatk/pull/3964. It basically bundles up everything under src/main/python/org/broadinstitute/hellbender/ into a zip file which is then pip installed as part of the conda env, along with all of the other dependencies.

The only reason that https://github.com/broadinstitute/gatk/pull/3964 currently says "DO NOT MERGE" is because I included a commit with a dummy python package, and a corresponding test, so I could validate that it works on travis. Let me know if you want to cherry-pick the core change, or we could merge that PR as is if you want to rebase on top of it.

So, I think most python code should live under src. You could also have tool-specific code that is kept as a tool resource to make it easy to call from Java, but that would be optional.

Reply:
OK, that all makes sense.  I’m fine with merging that PR and rebasing this one on top of it.

Reply:
Ok, its merged.

Reply:
I pushed a copy of this branch that is rebased on master to https://github.com/broadinstitute/gatk/tree/sl_gcnv_ploidy_cli_rebased to make sure we can use the changes by @cmnbroad mentioned above.  @mbabadi see the last commit in that branch for an example of how we might reorganize things.

I removed the duplicate copies of the gcnv/bin scripts (since these are already in src/main/resources), moved the gcnvkernel package to the same level as the gatkpython dummy package, and used find_packages() to install both as a single gatkpythonpackages package.  I retained a (slightly modified) copy of @mbabadi's original setup.py for gcnvkernel, which one can use to install that package separately should one choose to.

Not sure if this is how we were planning to organize things (and I must admit that I'm not familiar with the subtleties and conventions of python packaging), but it seems to work---we'll see if tests pass in that copy of the branch.

Reply:
Looks like tests run successfully in that branch, so what I did indeed works.  (Unfortunately, the tests don't pass, but that's because they are still running long...)

Reply:
I'll come back to this in a bit.

Reply:
quick note @cmnbroad: `argparse` is a core python3 package now and does not need to be installed separately.

Reply:
Oh ok, got it.